### DISCUSSION TITLE
Help with HTML formatting for email alerts

### DISCUSSION THREAD

**Discussion by lymanaukuleles** [jertel/elastalert2#821] [Q&A] [answered]
This is my rule file:

```
name: Lockout Rules File
type: any
num_events: 1
filter:
- query:
   query_string:
     query: "\"A user account was locked out\""
index: winlogbeat-*
realert:
 minutes: 0
query_key:
 - source.ip
include:
 - message
 - agent.hostname
 - source.ip
 - related.user
 - winlog.task
include_match_in_root: true
alert_subject: "Lockout Event Detected from {0}"
alert_subject_args:
 - related.user
alert_text:
<style type="text/css">
.tg  {border-collapse:collapse;border-spacing:0;}
.tg td{border-color:black;border-style:solid;border-width:1px;font-family:Arial, sans-serif;font-size:14px;
  overflow:hidden;padding:10px 5px;word-break:normal;}
.tg th{border-color:black;border-style:solid;border-width:1px;font-family:Arial, sans-serif;font-size:14px;
  font-weight:normal;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-78ue{background-color:#96fffb;border-color:#3166ff;color:#ecf4ff;text-align:center;vertical-align:top}
.tg .tg-0s4r{background-color:#ffffc7;border-color:inherit;color:#643403;text-align:left;vertical-align:top}
.tg .tg-23c7{background-color:#ffffc7;border-color:inherit;color:#3166ff;text-align:left;vertical-align:top}
.tg .tg-bolj{background-color:#ffccc9;border-color:inherit;text-align:center;vertical-align:top}
.tg .tg-0lax{text-align:left;vertical-align:top}
.tg .tg-psp6{background-color:#ffffc7;border-color:inherit;color:#00d2cb;text-align:left;vertical-align:top}
.tg .tg-6fx8{background-color:#ffffc7;border-color:inherit;color:#9b9b9b;text-align:left;vertical-align:top}
</style>
<table class="tg" style="undefined;table-layout: fixed; width: 792px">
<colgroup>
<col style="width: 162px">
<col style="width: 117px">
<col style="width: 111px">
<col style="width: 125px">
<col style="width: 277px">
</colgroup>
<thead>
  <tr>
    <th class="tg-78ue">Timestamp</th>
    <th class="tg-bolj">User</th>
    <th class="tg-bolj">Source IP</th>
    <th class="tg-bolj">Agent</th>
    <th class="tg-bolj">Message</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-0lax">{0}</td>
    <td class="tg-23c7">{1}</td>
    <td class="tg-psp6">{2}</td>
    <td class="tg-0s4r">{3}</td>
    <td class="tg-6fx8">{4}</td>
  </tr>
</tbody>
</table>
alert_text_args:
 - "@timestamp"
 - related.user
 - source.ip
 - agent.hostname
 - winlog.task
is_enabled: true
alert:
 - email:
    from_addr: "relay2@uhtasi.org"
    email: "info@uhtasi.org"
    smtp_host: "mail.uhtasi.org"
    smtp_port: 587
    smtp_ssl: false
    smtp_auth_file: "/etc/elastalert/smtp_auth_user.yaml"
alert_text_type: alert_text_only
_source_enabled: true
email_format: html
```
And this is the error:

`2022-04-20 23:53:14,503    DEBUG        elasticsearch > {"message":"Could not load rule /etc/elastalert/elastalert_rules/lockout.yaml: Could not parse file /etc/elastalert/elastalert_rules/lockout.yaml: while scanning a simple key\n  in \"<unicode string>\", line 24, column 1:\n    <style type=\"text/css\">\n    ^\ncould not find expected ':'\n  in \"<unicode string>\", line 25, column 1:\n    .tg  {border-collapse:collapse;b ... \n    ^","traceback":["Traceback (most recent call last):","  File \"/usr/local/lib/python3.10/site-packages/elastalert/loaders.py\", line 586, in get_yaml","    return read_yaml(filename)","  File \"/usr/local/lib/python3.10/site-packages/elastalert/yaml.py\", line 8, in read_yaml","    return yaml.load(yamlContent, Loader=yaml.FullLoader)","  File \"/usr/local/lib/python3.10/site-packages/yaml/__init__.py\", line 81, in load","    return loader.get_single_data()","  File \"/usr/local/lib/python3.10/site-packages/yaml/constructor.py\", line 49, in get_single_data","    node = self.get_single_node()","  File \"/usr/local/lib/python3.10/site-packages/yaml/composer.py\", line 36, in get_single_node","    document = self.compose_document()","  File \"/usr/local/lib/python3.10/site-packages/yaml/composer.py\", line 55, in compose_document","    node = self.compose_node(None, None)","  File \"/usr/local/lib/python3.10/site-packages/yaml/composer.py\", line 84, in compose_node","    node = self.compose_mapping_node(anchor)","  File \"/usr/local/lib/python3.10/site-packages/yaml/composer.py\", line 133, in compose_mapping_node","    item_value = self.compose_node(node, item_key)","  File \"/usr/local/lib/python3.10/site-packages/yaml/composer.py\", line 64, in compose_node","    if self.check_event(AliasEvent):","  File \"/usr/local/lib/python3.10/site-packages/yaml/parser.py\", line 98, in check_event","    self.current_event = self.state()","  File \"/usr/local/lib/python3.10/site-packages/yaml/parser.py\", line 449, in parse_block_mapping_value","    if not self.check_token(KeyToken, ValueToken, BlockEndToken):","  File \"/usr/local/lib/python3.10/site-packages/yaml/scanner.py\", line 115, in check_token","    while self.need_more_tokens():","  File \"/usr/local/lib/python3.10/site-packages/yaml/scanner.py\", line 152, in need_more_tokens","    self.stale_possible_simple_keys()","  File \"/usr/local/lib/python3.10/site-packages/yaml/scanner.py\", line 291, in stale_possible_simple_keys","    raise ScannerError(\"while scanning a simple key\", key.mark,","yaml.scanner.ScannerError: while scanning a simple key","  in \"<unicode string>\", line 24, column 1:","    <style type=\"text/css\">","    ^","could not find expected ':'","  in \"<unicode string>\", line 25, column 1:","    .tg  {border-collapse:collapse;b ... ","    ^","","During handling of the above exception, another exception occurred:","","Traceback (most recent call last):","  File \"/usr/local/lib/python3.10/site-packages/elastalert/elastalert.py\", line 1146, in load_rule_changes","    new_rule = self.rules_loader.load_configuration(rule_file, self.conf)","  File \"/usr/local/lib/python3.10/site-packages/elastalert/loaders.py\", line 223, in load_configuration","    rule = self.load_yaml(filename)","  File \"/usr/local/lib/python3.10/site-packages/elastalert/loaders.py\", line 242, in load_yaml","    loaded = self.get_yaml(filename)","  File \"/usr/local/lib/python3.10/site-packages/elastalert/loaders.py\", line 588, in get_yaml","    raise EAException('Could not parse file %s: %s' % (filename, e))","elastalert.util.EAException: Could not parse file /etc/elastalert/elastalert_rules/lockout.yaml: while scanning a simple key","  in \"<unicode string>\", line 24, column 1:","    <style type=\"text/css\">","    ^","could not find expected ':'","  in \"<unicode string>\", line 25, column 1:","    .tg  {border-collapse:collapse;b ... ","    ^"],"@timestamp":"2022-04-20T23:53:14.488794Z"}
`

Line 24/25 that is referenced in the error is right around where the HTML code starts. Basically just want to know a quick solution for this.

**Comment by jertel** [comment, marked as answer]
See the [YAML multiline syntax docs](https://yaml-multiline.info/). It should look something like this:

```
alert_text: |-
  <style
  ...
  </table>
```

Indents are important.
